starknet_devnet/starknet_wrapper.py

```diff
@@ -385,15 +385,20 @@ async def calculate_actual_fee(self, external_tx: InvokeFunction):
         )
```

Comment:
Does `call_info` contain any sign of failed execution? What does alpha-goerli return on `estimate_fee` on a contract that hasn't been deployed? Perhaps we should do the same.

Reply:
call data are fine as call info because in the devnet case actual fee token contract is deployed. 
If you call with an invalid contract address it will return
```
{
    "message": "Contract with address 0x62230ea046a9a5fbc261ac77d03c8d41e5d442db2284587570ab46455fd2489 is not deployed.",
    "status_code": 500
}
```
the problem here is not that contract is not deployed, but the block is not created.
on goerly there is no way there is no block. This is like edge case for devnet because you start it without any blocks.

Reply:
But is `estimate_fee` related to the fee token at all? We implement `estimate_fee` as a simulator of a function execution, but how can it be successfully executed if its contract is not yet deployed?

Reply:
Perhaps we can, before the simulation, extract from the request the target address and see if it's deployed.

Reply:
> But is `estimate_fee` related to the fee token at all? We implement `estimate_fee` as a simulator of a function execution, but how can it be successfully executed if its contract is not yet deployed?

If you call one of the contracts we pre-deploy then it can be executed, as state block number -1 has gass fee 0 this happend. Fee token has nothing to do with it except you can call it because it is predeployed.
This bug has nothing to do with contract being deployed or not:

- If the contract is not deployed response from my prev. answer will return and all will be fine
- if you try to call some of predeployed contracts zero division will happen as block -1 gas fee is 0